### PR TITLE
1.22 Feature Blog for Support for Windows privileged containers

### PR DESCRIPTION
**Pull request by kunal-kushwaha**
1.22 Feature Blog for "Support for Windows privileged containers" (https://github.com/kubernetes/enhancements/issues/1981)

Target publish date is ~16/08/21~ 2021-08-18.

cc @sftim @jlbutler @brasmith-ms

### CODE REVIEW

**Comment by kunal-kushwaha** [comment]
PR to be merged in this: https://github.com/kunal-kushwaha/website/pull/1

**Comment by tengqm** [comment]
Please don't forget to remove the `[WIP]` from the title when this one is ready for review/merge.

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+closed and the storage will be unmounted.
+
+HostProcess and Linux privileged containers enable similar scenarios but differ 
+greatly in their implementation, hence the naming difference. 
+
+{{< figure src="hostprocess-architecture.png" alt="HostProcess Architecture" >}}
+
+## How do I use it?
+
+HostProcess containers can be run from with in a
```
Comment (chrisnegus) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md.md:53]:
```suggestion
HostProcess containers can be run from within a 
```

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+[Create a Windows HostProcess Pod](/docs/tasks/configure-pod-container/create-hostprocess-container.md)
+
+[Pod Security Standards](/docs/concepts/security/pod-security-standards)
+
+[Windows HostProcess KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/1981-windows-privileged-container-support)
+
+## How do I get involved?
+
+HostProcess containers are in active development and welcome suggestions from the community.
```
Comment (chrisnegus) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md.md:70]:
```suggestion
HostProcess containers are in active development and we welcome suggestions from the community. 
```

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+{{<feature-state for_k8s_version="v1.22" state="alpha" >}}
+
+HostProcess containers aim to extend the Windows container model to enable a wider 
+range of Kubernetes cluster management scenarios. HostProcess containers run 
+directly on the host and maintain behavior and access similar to that of a regular 
+process. With HostProcess containers, users can package and distribute management 
+operations and functionalities that require host access while retaining versioning 
+and deployment methods provided by containers. This allows Windows containers to 
+be used for a variety of device plugin, storage, and networking management scenarios 
+in Kubernetes. With this comes the enablement of host network mode - allowing
```
Comment (chrisnegus) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md.md:19]:
```suggestion
in Kubernetes. With this comes the enablement of host network mode--allowing 
```

Comment (chrisnegus) [reply content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md.md:19]: The sentence above should not use a hyphen, so I changed it to a dash. A comma might actually be better. Perhaps someone else has an opinion.

**Review by chrisnegus — CHANGES_REQUESTED** [review]
The text looks good to me. I just had a couple of small editing suggestions.

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
+---
+layout: blog
+title: 'Windows HostProcess Containers'
+date: 2021-08-16
+slug: Windows HostProcess Containers
```
Comment (annajung) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md.md:5]: Hi, slug needs to change to not have spaces. please see an example here https://kubernetes.io/docs/contribute/new-content/blogs-case-studies/#technical-considerations-for-submitting-a-blog-post

**Review by sftim — COMMENTED** [review]
2 nits @kunal-kushwaha 
- the filename `content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md.md` has the file type duplicated
- also, the date in the filename doesn't match the publication date. Hugo doesn't mind but it can confuse us humans

plus some inline feedback

**Comment by sftim** [comment]
@kubernetes/sig-windows-misc: how's this article looking?

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+all the processes in the job will be signaled to exit, the job handle will be 
+closed and the storage will be unmounted.
+
+HostProcess and Linux privileged containers enable similar scenarios but differ 
+greatly in their implementation, hence the naming difference. 
+
+{{< figure src="hostprocess-architecture.png" alt="HostProcess Architecture" >}}
+
+## How do I use it?
+
```
[the remark this answers is not in the document]
Comment (kunal-kushwaha) [reply content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md.md]: cc @brasmith-ms

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 6]
```diff
@@ -0,0 +1,71 @@
+---
+layout: blog
+title: 'Alpha in v1.22: Windows HostProcess Containers'
+date: 2021-08-16
+slug: Windows-HostProcess-Containers
```
Comment (sftim) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:5]: nit:
```suggestion
slug: windows-hostprocess-containers
```

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+---
+
+**Authors:** Brandon Smith
```
Comment (sftim) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:8]:
```suggestion
**Authors:** Brandon Smith (Microsoft)
```

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+Kubernetes, including kube-proxy (via kubeadm), storage, and networking scenarios. 
+Support for these scenarios in Windows previously required workarounds via proxies 
+or other implementations. Using HostProcess containers, cluster operators no longer 
+need to log onto and individually configure each Windows node for administrative 
+tasks and management of Windows services. Operators can now utilize the container 
+model to deploy management logic to as many clusters as needed with ease. 
+
+## How does it work?
+
+Windows HostProcess containers are implemented with Job Objects, a break from the
```
Comment (sftim) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:35]:
```suggestion
Windows HostProcess containers are implemented with Windows _Job Objects_, a break from the
```

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+previous container model using server silos. Job objects provide the ability to 
+manage a group of processes as a group, and assign resource constraints to the 
+processes in the job. Job objects have no process or file system isolation, 
+enabling the privileged payload to view and edit the host file system with the 
+correct permissions, among other host resources. The init process, and any processes 
+it launches or that are explicitly launched by the user, are all assigned to the 
+job object of that container. When the init process exits or is signaled to exit, 
+all the processes in the job will be signaled to exit, the job handle will be 
+closed and the storage will be unmounted.
+
```
Comment (sftim) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:45]: Maybe mention that Job Objects, in the Windows sense, have nothing to do with Kubernetes' [Job](https://kubernetes.io/docs/concepts/workloads/controllers/job/) API.

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+HostProcess containers can be run from within a 
+[HostProcess pod](/docs/tasks/configure-pod-container/create-hostprocess-container.md). 
+With the feature enabled on Kubernetes version 1.22, a containerd container runtime of 
+1.5.4 or higher, and the latest version of hcsshim, deploying a pod spec with the 
+[correct HostProcess configuration](/docs/tasks/configure-pod-container/create-hostprocess-container.md#prerequisites) 
+will enable you to run HostProcess containers. 
+
+## How can I learn more?
+
+[Create a Windows HostProcess Pod](/docs/tasks/configure-pod-container/create-hostprocess-container.md)
```
Comment (sftim) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:62]:
```suggestion
- Work through [Create a Windows HostProcess Pod](/docs/tasks/configure-pod-container/create-hostprocess-container/)
```

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+[Pod Security Standards](/docs/concepts/security/pod-security-standards)
```
Comment (sftim) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:64]:
```suggestion
- Read about Kubernetes [Pod Security Standards](/docs/concepts/security/pod-security-standards/)
```

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+[Windows HostProcess KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/1981-windows-privileged-container-support)
```
Comment (sftim) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:66]:
```suggestion
- Read the enhancement proposal [Windows Privileged Containers and Host Networking Mode](https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/1981-windows-privileged-container-support) (KEP-1981)
```

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+
+## How do I get involved?
+
+HostProcess containers are in active development and we welcome suggestions from the community.
```
Comment (sftim) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:70]:
```suggestion
HostProcess containers are in active development. SIG Windows welcomes suggestions from the community.
```

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+[Create a Windows HostProcess Pod](/docs/tasks/configure-pod-container/create-hostprocess-container.md)
+
+[Pod Security Standards](/docs/concepts/security/pod-security-standards)
+
+[Windows HostProcess KEP](https://github.com/kubernetes/enhancements/tree/master/keps/sig-windows/1981-windows-privileged-container-support)
+
+## How do I get involved?
+
+HostProcess containers are in active development and we welcome suggestions from the community. 
+[Get involved with SIG-Windows to contribute!](https://github.com/kubernetes/community/tree/master/sig-windows)
```
Comment (sftim) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:71]: nit:
```suggestion
Get involved with [SIG Windows](https://github.com/kubernetes/community/tree/master/sig-windows)
to contribute!
```

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+{{<feature-state for_k8s_version="v1.22" state="alpha" >}}
+
+HostProcess containers aim to extend the Windows container model to enable a wider 
+range of Kubernetes cluster management scenarios. HostProcess containers run 
+directly on the host and maintain behavior and access similar to that of a regular 
+process. With HostProcess containers, users can package and distribute management 
+operations and functionalities that require host access while retaining versioning 
+and deployment methods provided by containers. This allows Windows containers to 
+be used for a variety of device plugin, storage, and networking management scenarios 
+in Kubernetes. With this comes the enablement of host network mode--allowing
```
Comment (sftim) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:19]: for an em dash, you need to use an actual em dash:
```suggestion
in Kubernetes. With this comes the enablement of host network mode—allowing
```

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 16]
```diff
@@ -0,0 +1,71 @@
+---
+layout: blog
+title: 'Alpha in v1.22: Windows HostProcess Containers'
+date: 2021-08-16
+slug: Windows-HostProcess-Containers
+---
+
+**Authors:** Brandon Smith
+
+{{<feature-state for_k8s_version="v1.22" state="alpha" >}}
```
Comment (sftim) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:10]: This is appropriate for docs. For a blog article, try:
```suggestion
Kubernetes v1.22 introduced a new alpha feature for clusters that
include Windows nodes: HostProcess containers.
```

**Review by sftim — DISMISSED** [review]
Thanks for this PR @brasmith-ms and @kunal-kushwaha 

I've suggested some changes; the important ones are:
1. Introduce this as a feature in text (don't use a `feature-state` shortcode)
2. explain the different between a (Windows) Job Object and a (Kubernetes) Job
3. fix the `slug` field in the front matter
4. Replace “we” with “SIG Windows”

Other feedback stands but IMO doesn't block publication.

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md (hunk 9 above)
Comment (kunal-kushwaha) [reply content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:46]: cc @brasmith-ms

**Comment by jlbutler** [comment]
Moving tracked publication date to Wednesday, Aug 18. Thanks @brasmith-ms !

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+processes in the job. Job objects have no process or file system isolation, 
+enabling the privileged payload to view and edit the host file system with the 
+correct permissions, among other host resources. The init process, and any processes 
+it launches or that are explicitly launched by the user, are all assigned to the 
+job object of that container. When the init process exits or is signaled to exit, 
+all the processes in the job will be signaled to exit, the job handle will be 
+closed and the storage will be unmounted.
+
+HostProcess and Linux privileged containers enable similar scenarios but differ 
+greatly in their implementation, hence the naming difference.
```
Comment (brasmith-ms) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:48]:
```suggestion
greatly in their implementation (hence the naming difference). HostProcess containers 
have their own pod security policies. Those used to configure Linux privileged 
containers **do not** apply. Enabling privileged access to a Windows host is a 
fundamentally different process than with Linux so the configuration and 
capabilities of each differ significantly. Below is a diagram detailing the 
overall architecture of Windows HostProcess containers:
```

**Review by brasmith-ms — CHANGES_REQUESTED** [review]

**Comment by kunal-kushwaha** [comment]
Thanks @brasmith-ms!

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md (hunk 9 above)
Comment (brasmith-ms) [reply content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:46]: Updated!

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 18]
```diff
@@ -0,0 +1,78 @@
...
+{{< figure src="hostprocess-architecture.png" alt="HostProcess Architecture" >}}
+
+## How do I use it?
+
+HostProcess containers can be run from within a 
+[HostProcess pod](/docs/tasks/configure-pod-container/create-hostprocess-container.md). 
+With the feature enabled on Kubernetes version 1.22, a containerd container runtime of 
+1.5.4 or higher, and the latest version of hcsshim, deploying a pod spec with the 
+[correct HostProcess configuration](/docs/tasks/configure-pod-container/create-hostprocess-container.md#prerequisites) 
+will enable you to run HostProcess containers.
```
Comment (brasmith-ms) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:64]:
```suggestion
will enable you to run HostProcess containers. To get started with running 
Windows containers see the general guidance for [Windows in Kubernetes](/docs/setup/production-environment/windows/)
```

**Review by brasmith-ms — CHANGES_REQUESTED** [review]

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (brasmith-ms) [reply content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md.md]: Updated!

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 19]
```diff
@@ -0,0 +1,79 @@
...
+[HostProcess pod](/docs/tasks/configure-pod-container/create-hostprocess-container.md). 
+With the feature enabled on Kubernetes version 1.22, a containerd container runtime of 
+1.5.4 or higher, and the latest version of hcsshim, deploying a pod spec with the 
+[correct HostProcess configuration](/docs/tasks/configure-pod-container/create-hostprocess-container.md#prerequisites) 
+will enable you to run HostProcess containers. To get started with running 
+Windows containers see the general guidance for [Windows in Kubernetes](/docs/setup/production-environment/windows/)
+
+## How can I learn more?
+
+- Work through [Create a Windows HostProcess Pod](/docs/tasks/configure-pod-container/create-hostprocess-container/)
```
Comment (sftim) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:69]: I just spotted this and will Just Fix It as it's a mechanical fix
```suggestion
- Work through [Create a Windows HostProcess Pod](/docs/tasks/configure-pod-container/create-hostprocess-pod/)
```

**Review by sftim — COMMENTED** [review]
/approve
Ready for an LGTM!

**Comment by sftim** [comment]
/label tide/merge-method-squash

@kunal-kushwaha you're also welcome to do a squash-and-force-push

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 20]
```diff
@@ -0,0 +1,79 @@
...
+include Windows nodes: HostProcess containers.
+
+HostProcess containers aim to extend the Windows container model to enable a wider 
+range of Kubernetes cluster management scenarios. HostProcess containers run 
+directly on the host and maintain behavior and access similar to that of a regular 
+process. With HostProcess containers, users can package and distribute management 
+operations and functionalities that require host access while retaining versioning 
+and deployment methods provided by containers. This allows Windows containers to 
+be used for a variety of device plugin, storage, and networking management scenarios 
+in Kubernetes. With this comes the enablement of host network mode—allowing
```
Comment (jsturtevant) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:20]:
```suggestion
in Kubernetes. With this comes the enablement of host network mode — allowing
```

**Review by jsturtevant — COMMENTED** [review]
content looks good. I saw one typo, not sure if it would be considered blocking

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 21]
```diff
@@ -0,0 +1,79 @@
...
+fundamentally different process than with Linux so the configuration and 
+capabilities of each differ significantly. Below is a diagram detailing the 
+overall architecture of Windows HostProcess containers:
+
+{{< figure src="hostprocess-architecture.png" alt="HostProcess Architecture" >}}
+
+## How do I use it?
+
+HostProcess containers can be run from within a 
+[HostProcess pod](/docs/tasks/configure-pod-container/create-hostprocess-container.md).
```
Comment (reylejano) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:60]:
```suggestion
[HostProcess Pod](/docs/tasks/configure-pod-container/create-hostprocess-pod.md). 
```
Broken link

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 22]
```diff
@@ -0,0 +1,79 @@
...
+
+{{< figure src="hostprocess-architecture.png" alt="HostProcess Architecture" >}}
+
+## How do I use it?
+
+HostProcess containers can be run from within a 
+[HostProcess pod](/docs/tasks/configure-pod-container/create-hostprocess-container.md). 
+With the feature enabled on Kubernetes version 1.22, a containerd container runtime of 
+1.5.4 or higher, and the latest version of hcsshim, deploying a pod spec with the 
+[correct HostProcess configuration](/docs/tasks/configure-pod-container/create-hostprocess-container.md#prerequisites)
```
Comment (reylejano) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:63]:
```suggestion
[correct HostProcess configuration](/docs/tasks/configure-pod-container/create-hostprocess-pod/#before-you-begin) 
```
@kunal-kushwaha Is this the correct link?

**Comment by sftim** [comment]
I'm going to commit those hyperlink fixes. Still looking for LGTM.

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md (hunk 20 above)
Comment (sftim) [reply content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:20]: I'm happy with either punctuation.

**Comment by Atharva-Shinde** [comment]
I thoroughly read the article and found no typos or broken links.

**Comment by Atharva-Shinde** [comment]
LGTM !!!!

**Comment by Atharva-Shinde** [comment]
I thoroughly read the article and found no typos or broken links :)

File: content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md [hunk 23]
```diff
@@ -0,0 +1,79 @@
...
+directly on the host and maintain behavior and access similar to that of a regular 
+process. With HostProcess containers, users can package and distribute management 
+operations and functionalities that require host access while retaining versioning 
+and deployment methods provided by containers. This allows Windows containers to 
+be used for a variety of device plugin, storage, and networking management scenarios 
+in Kubernetes. With this comes the enablement of host network mode—allowing
+HostProcess containers to be created within the host's network namespace instead of 
+their own. HostProcess containers can also be built on top of existing Windows server 
+2019 (or later) base images, managed through the Windows container runtime, and run 
+as any user that is available on or in the domain of the host machine.
```
Comment (dcantah) [inline content/en/blog/_posts/2021-08-11-support-for-HostProcess-Containers/index.md:24]: This isn't entirely true at the moment

**Pull request merged**